corehq/apps/reports/tests/test_case_list_explorer_report.py

```diff
@@ -0,0 +1,82 @@
+import json
+from django.http.request import QueryDict
+from django.test import TestCase
+from django.test.client import RequestFactory
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.reports.standard.cases.case_list_explorer import CaseListExplorer
+from corehq.apps.users.models import (
+    DomainMembership,
+    WebUser,
+)
+from corehq.util.test_utils import flag_enabled
+
+
+class TestCaseListExplorer(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain_name = 'case-list-test'
+        cls.domain = Domain.get_or_create_with_name(cls.domain_name, is_active=True)
+        cls.user = WebUser(username='test@cchq.com', domains=[cls.domain_name], is_admin=True)
+        cls.user.domain_memberships.append(DomainMembership(domain=cls.domain_name, role_id='admin'))
+        cls.request_factory = RequestFactory()
+
+    def setUp(self):
+        super().setUp()
+        self.request = self.request_factory.get('/some/url')
+        self.request.couch_user = self.user
+        self.request.domain = self.domain
+        self.request.can_access_all_locations = True
```

Comment:
oh wow, that needed quite the setup. How about using `corehq.apps.users.tests.test_auditors.MockRequest`?

Reply:
I tried `MockRequest` it does not replicate whole request interface and lacks http `method`, so `request.GET` doesn't work with it.

Reply:
How hard would it be to modify `MockRequest` to support what you need?
This would help others to use `MockRequest` if they need to mock requests and avoid re-inventing the wheel each time.

Reply:
The underlying view is expecting a whole request object, hence the changes here don't seem feasible to add to the `MockRequest` class, `RequestFactory` generates a request object for testing so it can be changed accordingly for requirements of the test.